plugins/layout/src/main/java/org/qbicc/plugin/layout/Layout.java

```diff
@@ -45,6 +46,18 @@ public static Layout get(CompilationContext ctxt) {
         return layout;
     }
 
+    public static Layout getForInterpreter(CompilationContext ctxt) {
```

Comment:
Is the separate layout for the interpreter just used for static fields?  Or is it used to account for differences in the instance fields between qbicc and hostVM classes (where such differences exist)?

Reply:
The difference here is that it's possible for fields to be injected that the interpreter does not know about because the interpreter may need the layout info at an earlier point than would have otherwise been determined. It's possible we could reconcile this at some point though - I've ended up needing to force the injection of most of the interesting injected fields earlier anyway.

The static fields layout method is only used by the interpreter and the results are not cached so the actual Layout instance is not of particular importance for this use case.

Reply:
thanks.  Makes sense.